plugins/notify/__init__.py

```diff
@@ -14,124 +18,335 @@
-ATTACH_COVERS_OPTION_ALLOWED = False
+# For documentation on libnotify see also the "Desktop Notifications Specification":
+# https://developer.gnome.org/notification-spec/
+gi.require_version('Notify', '0.7')
+from gi.repository import Notify
+
 
-Notify.init('exailenotify')
+LOGGER = logging.getLogger(__name__)
+DEFAULT_ICON_SIZE = (48, 48)
 
+BODY_ARTIST_ALBUM = _('from {album} by {artist}')
+BODY_ARTIST = _('by {artist}')
+BODY_ALBUM = _('by {album}')
 
-class ExaileNotification(object):
 
-    def __init__(self):
-        self.notification_id = None
-        self.exaile = None
+class NotifierSettings(object):
 
     def __inner_preference(klass):
         """Function will make a property for a given subclass of Preference"""
 
         def getter(self):
-            return settings.get_option(klass.name, klass.default or None)
+            return xl_settings.get_option(klass.name, klass.default or None)
 
         def setter(self, val):
-            settings.set_option(klass.name, val)
+            xl_settings.set_option(klass.name, val)
+
+        # Migration:
+        if hasattr(klass, 'pre_migration_name'):
+            if xl_settings.get_option(klass.name, None) is None:
+                old_value = xl_settings.get_option(
+                    klass.pre_migration_name, klass.default or None)
+                if old_value is not None:
+                    xl_settings.set_option(klass.name, old_value)
 
         return property(getter, setter)
 
-    resize = __inner_preference(notifyprefs.ResizeCovers)
-    body_artistalbum = __inner_preference(notifyprefs.BodyArtistAlbum)
-    body_artist = __inner_preference(notifyprefs.BodyArtist)
-    body_album = __inner_preference(notifyprefs.BodyAlbum)
-    summary = __inner_preference(notifyprefs.Summary)
-    attach_tray = __inner_preference(notifyprefs.AttachToTray)
+    notify_pause = __inner_preference(notifyprefs.NotifyPause)
+    resize_covers = __inner_preference(notifyprefs.ResizeCovers)
+    show_covers = __inner_preference(notifyprefs.ShowCovers)
+    show_when_focused = __inner_preference(notifyprefs.ShowWhenFocused)
+    tray_hover = __inner_preference(notifyprefs.TrayHover)
+    use_media_icons = __inner_preference(notifyprefs.UseMediaIcons)
 
-    @common.threaded
-    def on_play(self, type, player, track):
-        '''Callback when we want to display a notification
 
-        type and player arguments are ignored.
+class Notifier(PlaybackAdapter):
+
+    settings = NotifierSettings()
+
+    def __init__(self, exaile, caps):
+        self.__exaile = exaile
+        self.__old_icon = None
+        self.__tray_connection = None
+
+        notification = Notify.Notification.new("", None, None)
+
+        if 'sound' in caps:
+            notification.set_hint('suppress-sound',
+                                  GLib.Variant.new_boolean(True))
+        # notification.set_hint('suppress-sound', True)
+        self.settings.can_show_markup = 'body-markup' in caps
```

Comment:
This is the place where I detect whether the notification server supports markup or not.